src/bin/server.rs

```diff
@@ -484,6 +550,69 @@ fn main() {
                                 }
                                 Request::AdvanceInput(name, tx) => server.advance_input(name, tx),
                                 Request::CloseInput(name) => server.close_input(name),
+                                #[cfg(feature="graphql")]
+                                Request::GraphQl(name, query) => {
+                                    if owner == worker.index() {
+                                        // we are the owning worker and thus have to
+                                        // keep track of this client's new interest
+
+                                        match command.client {
+                                            None => {}
+                                            Some(client) => {
+                                                let client_token = Token(client);
+                                                server.interests
+                                                    .entry(name.clone())
+                                                    .or_insert(Vec::new())
+                                                    .push(client_token);
+                                            }
+                                        }
+                                    }
+
+                                    if !server.context.global_arrangements.contains_key(&name) {
+
+                                        let send_results_handle = send_results.clone();
+
+                                        worker.dataflow::<u64, _, _>(|scope| {
+                                            server.register_graph_ql(query, &name, scope);
+
+                                            server
+                                                .interest(&name, scope)
+                                                .import_named(scope, &name)
+                                                .as_collection(|tuple,_| tuple.clone())
```

Comment:
It should be possible to use `.import_named().stream` here directly, instead of converting to collection and back again.

Reply:
It is not, in fact!